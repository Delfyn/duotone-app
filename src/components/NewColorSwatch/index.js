/* @flow */

import React from 'react';
import './style.css';

type Props = {
  className?: string,
  size?: number,
  rotation?: number,
  onClick: Function,
};

const NewColorSwatch = ({
  className = '',
  size = 40,
  rotation = 0,
  onClick,
}: Props) => (
  <svg
    className={className}
    onClick={onClick}
    width={size}
    height={size}
    viewBox={`0 0 ${size} ${size}`}
    version="1.1"
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink">
    <defs>
      <circle id="path-1" cx={size / 2} cy={size / 2} r={size / 2} />
      <mask
        id="mask-2"
        maskContentUnits="userSpaceOnUse"
        maskUnits="objectBoundingBox"
        x="0"
        y="0"
        width={size}
        height={size}
        fill="white">
        <use xlinkHref="#path-1" />
      </mask>
      <polygon
        id="path-3"
        points="21 19 21 11 19 11 19 19 11 19 11 21 19 21 19 29 21 29 21 21 29 21 29 19"
      />
      <mask
        id="mask-4"
        maskContentUnits="userSpaceOnUse"
        maskUnits="objectBoundingBox"
        x="0"
        y="0"
        width={size / 2}
        height={size / 2}
        fill="white">
        <use xlinkHref="#path-3" />
      </mask>
    </defs>
    <g
      id="Page-1"
      stroke="none"
      strokeWidth="1"
      fill="none"
      fillRule="evenodd"
      transform={`rotate(${rotation} ${size / 2} ${size / 2})`}>
      <g id="Artboard-2" stroke="#979797">
        <g id="Plus">
          <g id="Oval-3">
            <use mask="url(#mask-2)" strokeWidth="4" xlinkHref="#path-1" />
            <use
              id="Combined-Shape"
              mask="url(#mask-4)"
              strokeWidth="2"
              fill="#D8D8D8"
              fillRule="evenodd"
              xlinkHref="#path-3"
            />
          </g>
        </g>
      </g>
    </g>
  </svg>
);

export default NewColorSwatch;
