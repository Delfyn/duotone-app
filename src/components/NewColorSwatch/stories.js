/* @flow */

import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import NewColorSwatch from './index';

storiesOf('NewColorSwatch', module).add('Simple example', () => (
  <NewColorSwatch onClick={action('onClick')} />
));
