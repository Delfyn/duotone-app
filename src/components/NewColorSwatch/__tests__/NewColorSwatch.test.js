import React from 'react';
import renderer from 'react-test-renderer';
import NewColorSwatch from '../index';

test('NewColorSwatch renders correctly', () => {
  const component = renderer.create(
    <NewColorSwatch />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
