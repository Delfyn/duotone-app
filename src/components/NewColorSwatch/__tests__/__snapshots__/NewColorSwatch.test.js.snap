// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewColorSwatch renders correctly 1`] = `
<svg
  className=""
  height={40}
  onClick={undefined}
  version="1.1"
  viewBox="0 0 40 40"
  width={40}
  xmlns="http://www.w3.org/2000/svg"
  xmlnsXlink="http://www.w3.org/1999/xlink"
>
  <defs>
    <circle
      cx={20}
      cy={20}
      id="path-1"
      r={20}
    />
    <mask
      fill="white"
      height={40}
      id="mask-2"
      maskContentUnits="userSpaceOnUse"
      maskUnits="objectBoundingBox"
      width={40}
      x="0"
      y="0"
    >
      <use
        xlinkHref="#path-1"
      />
    </mask>
    <polygon
      id="path-3"
      points="21 19 21 11 19 11 19 19 11 19 11 21 19 21 19 29 21 29 21 21 29 21 29 19"
    />
    <mask
      fill="white"
      height={20}
      id="mask-4"
      maskContentUnits="userSpaceOnUse"
      maskUnits="objectBoundingBox"
      width={20}
      x="0"
      y="0"
    >
      <use
        xlinkHref="#path-3"
      />
    </mask>
  </defs>
  <g
    fill="none"
    fillRule="evenodd"
    id="Page-1"
    stroke="none"
    strokeWidth="1"
    transform="rotate(0 20 20)"
  >
    <g
      id="Artboard-2"
      stroke="#979797"
    >
      <g
        id="Plus"
      >
        <g
          id="Oval-3"
        >
          <use
            mask="url(#mask-2)"
            strokeWidth="4"
            xlinkHref="#path-1"
          />
          <use
            fill="#D8D8D8"
            fillRule="evenodd"
            id="Combined-Shape"
            mask="url(#mask-4)"
            strokeWidth="2"
            xlinkHref="#path-3"
          />
        </g>
      </g>
    </g>
  </g>
</svg>
`;
